### DISCUSSION TITLE
Mesh contains holes after using the filter boolean_union

### DISCUSSION THREAD

**Discussion by JanaBrunatova** [pyvista/pyvista#2134] [Q&A]
Hi everyone, 

I am trying to compare two meshes in terms of volumetric change. It is easy to compute the volume of each one (closed surface) mesh, but my wish is to compute boolean union and difference of two (surface) meshes, then close them and compute corresponding volumes. To do this, I used `boolean_union` and filter `fill_holes`. Unfortunately, the resulting mesh looks horrible and still contains some holes. Does anyone have some idea how to improve the procedure or (at least) the resulting mesh?

Thanks in advance, 

Jana

![image](https://user-images.githubusercontent.com/98927648/152264144-c96de5ba-2d7b-4901-93f8-2fd4055b1260.png)

**Comment by banesullivan** [comment]
@janca42, this is a generally tough problem to solve. Would you be able to share your mesh files and your current code?

**Comment by JanaBrunatova** [comment]
Of course, @banesullivan, my current code, input and output meshes are included here:
[surface_boolean_operations.zip](https://github.com/pyvista/pyvista/files/8022492/surface_boolean_operations.zip)

If you run the script `surf_boolean_operations.py`, the (incorrect) output volumes will be printed. Both intersection and union volumes are above 1000 mm^3. These numbers cannot be right since `mesh2.stl` has approximately a volume of 450 mm^3. I hope that if I could somehow fill the holes in the surface meshes, the volume filters should give me the correct numbers, both for the intersection and union. (The union surface is much worse, but the intersection also contains at least one hole.)

Thank you very much for your time. 

I would appreciate any advice. :) 
Jana
